proc/cut.go

```diff
@@ -27,96 +34,163 @@ type Cut struct {
 // do this now since it might confuse users who expect to see output
 // fields in the order they specified.
 func CompileCutProc(c *Context, parent Proc, node *ast.CutProc) (*Cut, error) {
-	resolvers, err := expr.CompileFieldExprs(node.Fields)
+	var fields []string
+	for _, field := range node.Fields {
+		fields = append(fields, expr.FieldExprToString(field))
+	}
```

Comment:
Seems like between here and groupby, we have enough cases where procs operate on dotted field names that it would be nice to have the parser just put the full dotted field name into the ast rather than working backward from a tree of FieldExprs here.
That could easily be done in a follow-up though.

Reply:
Agreed. I think this should be done outside of this PR though. Will file an issue.

Reply:
https://github.com/brimsec/zq/issues/643